### PR TITLE
Pad op rewrite

### PR DESCRIPTION
**Pull request by kahmed10**

### CODE REVIEW

**Comment by pfultz2** [comment]
In general, I would like all of our passes to start with a verb(ie `eliminate_concat`) rather than ending with a verb(ie `dead_code_elimination`). At some point, I need to go back and rename the old ones. For now, at least, I try to name the new passes in this way.

Also, I would prefer it be called `eliminate_pad` instead of `rewrite_pad`.

File: test/eliminate_identity_test.cpp [hunk 1]
```diff
@@ -59,7 +59,7 @@ TEST_CASE(simple_test_end_dependency)
     p.add_instruction(sum_op{}, ans, three);
     p.add_instruction(migraphx::op::identity{}, ans);
     p.compile(eliminate_identity_target{});
-    EXPECT(!std::none_of(p.begin(), p.end(), [](const migraphx::instruction& ins) {
+    EXPECT(std::any_of(p.begin(), p.end(), [](const migraphx::instruction& ins) {
```
Comment (pfultz2) [inline test/eliminate_identity_test.cpp]: Shouldn't this be in a separate PR?

File: src/include/migraphx/operators.hpp [hunk 2]
```diff
@@ -714,6 +714,17 @@ struct pad
...
     }
+
+    bool symmetric() const
+    {
+        std::size_t num_dims = pads.size() / 2;
+        for(std::size_t i = 0; i < num_dims; i++)
+        {
+            if(pads.at(i) != pads.at(i + num_dims))
+                return false;
+        }
```
Comment (pfultz2) [inline src/include/migraphx/operators.hpp]: This could be written as `std::equal(pads.begin(), pads.begin()+num_dims, pads.begin()+num_dims, pads.end())`.

File: test/eliminate_identity_test.cpp (hunk 1 above)
Comment (kahmed10) [reply test/eliminate_identity_test.cpp]: There were some changes I made to #213 after it was closed that were very minor. I can separate if necessary.

**Comment by mvermeulen** [comment]
Note, with this change I do observe a speedup for inceptionv3 model exported from PyTorch 1.0.  Here is the overall summary for inceptionv3, batch size 32:

- With PR 219, PyTorch 1.0 = 98.9458ms
- With PR 219, PyTorch 0.4 = 98.9705ms
- Without PR 219, PyTorch 1.0 = 110.013ms
- Without PR 219, PyTorch 0.4 = 98.9988ms

Note: Inception v3 is the model that saw the largest change in going to PyTorch 1.0

File: test/eliminate_identity_test.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (kahmed10) [reply test/eliminate_identity_test.cpp]: changed in a different PR
